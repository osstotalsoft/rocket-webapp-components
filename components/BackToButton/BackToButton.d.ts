import { ButtonProps } from '@material-ui/core/Button';

export interface BackToButtonProps extends Omit<ButtonProps, 'color'> {
  /**
  * The title of button.
  */
  title?: string
  /**
  * @default "theme"
  * The color of button.
  */
  color?: "primary" |
  "info" |
  "theme" |
  "themeNoBackground" |
  "themeWithBackground" |
  "success" |
  "warning" |
  "danger" |
  "rose" |
  "white" |
  "simple" |
  "defaultNoBackground" |
  "primaryNoBackground" |
  "infoNoBackground" |
  "successNoBackground" |
  "warningNoBackground" |
  "dangerNoBackground" |
  "roseNoBackground"
  /**
  * @default "medium"
  * The size of button.
  */
  size?: "medium" | "small"
  /**
  * @default "small"
  * The fontsize of button's content.
  */
  fontSize?: "inherit" | "small" | "medium" | "large"
  /**
  * The path where the browser will be directed to when the button is clicked.
  */
  path?: string
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/back-to-button
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/BackToButton/BackToButton.js
 * 
 * Material-UI component used:
 * 
 * - IconButton: https://material-ui.com/api/icon-button/
 * 
 */
export default function BackToButton(props: BackToButtonProps): JSX.Element;