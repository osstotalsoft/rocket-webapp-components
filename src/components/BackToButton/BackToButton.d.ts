import { ButtonProps } from '@material-ui/core/Button';

export interface BackToButtonProps extends Omit<ButtonProps, 'color'> {
  /**
  * The title of button.
  */
  title?: string
  /**
  * @default "theme"
  * The color of button.
  */
  color?: "primary" |
  "info" |
  "theme" |
  "themeNoBackground" |
  "themeWithBackground" |
  "success" |
  "warning" |
  "danger" |
  "rose" |
  "white" |
  "simple" |
  "defaultNoBackground" |
  "primaryNoBackground" |
  "infoNoBackground" |
  "successNoBackground" |
  "warningNoBackground" |
  "dangerNoBackground" |
  "roseNoBackground" |
  "themeWithGradient"
  /**
  * @default "medium"
  * The size of button.
  */
  size?: "medium" | "small"
  /**
  * @default "small"
  * The fontsize of button's content.
  */
  fontSize?: "inherit" | "small" | "medium" | "large"
  /**
  * The path where the browser will be directed to when the button is clicked.
  */
  path?: string
  /**
  * Override or extend the styles applied to the component
  */
  className?: string
  /**
  * @default false
  * If true, button will be disabled.
  */
  disabled?: boolean
  /**
  * @default false
  * If true, the keyboard focus ripple will be disabled. 
  */
  disableFocusRipple?: boolean
  /**
  * If true, the ripple effect will be disabled.
  */
  disableRipple?: boolean
  /**
  * @default "false"
  * If given, uses a negative margin to counteract the padding on one side 
  * (this is often helpful for aligning the left or right side of the icon 
  * with content above or below, without ruining the border size and shape).
  */
  edge?: "start" | "end" | "false"
  
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/back-to-button
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/BackToButton/BackToButton.js
 * 
 * Bit component used:
 * 
 * - IconButton: https://bit.dev/totalsoft_oss/react-mui/icon-button
 * 
 */
export default function BackToButton(props: BackToButtonProps): JSX.Element;