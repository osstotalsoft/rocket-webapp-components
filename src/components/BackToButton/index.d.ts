export { default } from './BackToButton';
export * from './BackToButton';