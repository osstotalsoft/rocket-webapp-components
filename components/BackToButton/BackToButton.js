import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { useHistory } from 'react-router-dom';
import { ArrowBack } from '@material-ui/icons';
import CustomIconButton from '../IconButton';

function BackToButton({ title, path, fontSize, ...rest }) {
    const history = useHistory();

    const onBackToList = useCallback(() => {
        history.push(path);
    }, [history, path])

    return (
        <CustomIconButton aria-label="Back" onClick={onBackToList} tooltip={title} {...rest}>
            <ArrowBack fontSize={fontSize} />
        </CustomIconButton>
    )
}

BackToButton.defaultProps = {
    color: "theme",
    fontSize: "small",
    size: "medium"
};

BackToButton.propTypes = {
    /**
  * The title of button.
  */
    title: PropTypes.string,
    /**
   * The color of button.
   */
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground",
        "themeWithGradient"
    ]),
    /**
      * The size of button.
      */
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    /**
  * The fontSize of button's content.
  */
    fontSize: PropTypes.oneOf([
        "inherit",
        "small",
        "medium",
        "large"
    ]),
    /**
  * The path where the browser will be directed to when the button is clicked.
  */
    path: PropTypes.string
}

export default BackToButton;