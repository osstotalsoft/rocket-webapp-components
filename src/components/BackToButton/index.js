import BackToButton from "./BackToButton";
export default BackToButton;