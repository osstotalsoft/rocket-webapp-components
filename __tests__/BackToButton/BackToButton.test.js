import { mount } from 'enzyme';
import React from 'react';
import BackToButton from '../../components/BackToButton/BackToButton';

describe('BackToButton', () => {
    it('the color, fontSize and size props should be equal to the default values if they are not provided', () => {

        const wrapper = mount(<BackToButton
            title="Back"
        ></BackToButton>);

        expect(wrapper.props().color).toBe('theme');
        expect(wrapper.props().fontSize).toBe('small');
        expect(wrapper.props().size).toBe('medium');
    });
});