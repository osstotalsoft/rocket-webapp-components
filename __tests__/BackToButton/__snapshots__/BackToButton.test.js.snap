// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToButton snapshot 1`] = `
<span
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Back"
>
  <button
    aria-label="Back"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</span>
`;
